frontend/src/components/utils/Search.tsx

```diff
@@ -182,7 +182,10 @@ const Search: React.FC<Props> = React.memo(({ dashboard }) => {
   const { t } = useTranslation();
   const track = useAnalyticsTrack();
 
-  const [value, setValue] = useQueryParam("query");
+  const [queryValue, setQueryValue] = useQueryParam("query");
+  const [inputValue, setInputValue] = React.useState<string | undefined>(
```

Comment:
Sorry I didn't make a thorough review in one go, but I think we can do 2 more things here:
1. Make `inputValue` always be a string - (`React.useState(queryValue || "")`), otherwise react will complain that element is changing state between being controllable and uncontrollable.
2. Use only one setter for `onChange` handler - `setInputValue`, and update query via `useEffect` as it is actually an effect - query string should update as an effect of a input value change.

I'll merge right after those are fulfilled.

@ruziev-dev Appreciate your effort a lot!